### PR TITLE
♿ TetraLogical: AMP Stories accessibility review / documentation

### PR DESCRIPTION
**Pull request by caroqliu**
This PR is a copy of #5397 with the following changes:
- Fresh merge to `future`
- Addresses most comments on original PR, adding links to ampproject/amphtml#24821 but not ampproject/amphtml#32751 which has since been closed.

/to original PR reviewers @newmuis @mszylkowski 

Original PR description: 
> Suggested changes and additions based on the TetraLogical accessibility review commissioned by @nainar / @caroqliu
>
> Main changes:
>
> - added a few minor clarifications, where appropriate, advising authors to consider use information conveyed through audio, judicious use/avoiding overuse of animations, suggested contrast ratios for text over backgrounds.
> - reworked the explanation of fallback content - not just as a way to show a "your browser doesn't support this", but an opportunity to still convey useful information to users
> - emphasised that allowing landscape mode (rather than just showing a door-slam "tilt your device" type message) is preferable (as this would otherwise contravene https://www.w3.org/TR/WCAG/#orientation ... but probably something that AMP Stories should address at a more fundamental/systemic level, as noted in the recent accessibility review)
> - fixes to visually broken pages (non-working "Develop > Animate & Transition" in the left-hand navigation, missing left-hand sidebar in https://amp.dev/documentation/guides-and-tutorials/optimize-and-measure/discovery_web_stories/?format=stories, broken bottom part of the page in https://amp.dev/documentation/guides-and-tutorials/develop/style_and_layout/responsive_design/?format=stories)
> - rewritten text alternatives for meaningful images/illustrations in the documentation where necessary
> - added placeholder alt="..." to most code sample examples of images, as this increases the chance that authors will actually remember to use the attribute to provide text alternatives
> - small fixes to typos found along the way
>
>x-ref: ampproject/amphtml#32779

### CODE REVIEW

**Review by sebastianbenz — APPROVED** [review]
Thanks!

File: examples/source/1.components/amp-story-360.html [hunk 1]
```diff
@@ -113,7 +115,7 @@
             duration="3s"
             controls="gyroscope">
             <amp-img
-              src="/static/samples/img/amp-story-360.jpg" width="4300" height="2150">
+              src="/static/samples/img/amp-story-360.jpg" width="4300" height="2150" alt="..."">
```
[the remark this answers is not in the document]
Comment (caroqliu) [reply examples/source/1.components/amp-story-360.html:118]: Good catch, fixed!

File: examples/source/1.components/amp-story-interactive-poll/index.html [hunk 2]
```diff
@@ -192,7 +196,11 @@
 </html>
 
 <!--
-        # More resources
+  ## Accessibility considerations
 
-        If you want to learn more about all the interactive experiences for Web Stories, head over to the [documentation](/documentation/components/amp-story-interactive?format=stories) or check out the guide on how to make a [story with quizzes](/documentation/examples/components/amp-story-interactive-quiz/?format=stories) instead.
-      -->
+  Currently, this component lacks keyboard and screen reader accessibility.
```
[the remark this answers is not in the document]
Comment (caroqliu) [reply examples/source/1.components/amp-story-interactive-poll/index.html:201]: Awesome, I removed this line but LMK if you think the entire section can be removed.

[the remark this answers is not in the document]
Comment (caroqliu) [reply examples/source/1.components/amp-story-interactive-poll/index.html:201]: Removed!

File: examples/source/style-layout/Desktop_and_Landscape_Mode_Support.html [hunk 3]
```diff
@@ -36,6 +36,7 @@
     <!-- If `supports-landscape` attribute is specified on the `amp-story` element, it will:
     - allow the story to be seen when a mobile device is held in a landscape orientation
     - change the desktop experience to an immersive full bleed mode, replacing the default three portrait panels experience -->
+    <!-- [tip type="note"]While this is currently opt-in and optional, we strongly recommend making sure that users on mobile devices are able to view stories in whatever orientation best suits their needs - otherwise, they will simply be presented with a "The page is best viewed in portrait mode" message.[/tip] -->
```
[the remark this answers is not in the document]
Comment (caroqliu) [reply examples/source/style-layout/Desktop_and_Landscape_Mode_Support.html:39]: Removed for now.

File: examples/source/visual-effects/AMP_Story_Animations.html [hunk 4]
```diff
@@ -8,6 +8,8 @@
   ## Introduction
 
   AMP Stories support animations out-of-the-box.  Every element inside an `amp-story-page` can define an entrance animation. You can find a list of all supported animations [here](/content/amp-dev/documentation/components/reference/amp-story.md#animations).
+
+  [tip type="note"]Animations can help make your Web Story more visually exciting and engaging, but use them sparingly. Some users may find long, continuous animations distracting. Other users may have motion sensitivity and be adversely affected by excessive use of motion and parallax effects. This [tracking issue](https://github.com/ampproject/amphtml/issues/24821) follows the effort to respect the `prefers-reduced-motion` directive.[/tip]
```
[the remark this answers is not in the document]
Comment (caroqliu) [reply examples/source/visual-effects/AMP_Story_Animations.html:12]: Updated wording and removed link, PTAL

File: examples/source/visual-effects/Ken_Burns.html [hunk 5]
```diff
@@ -63,12 +63,14 @@
     - Animate the `<amg-img>` element using the `animate-in="pan-right"`. (You can also use `pan-left`, `pan-up`, and `pan-down`). [See amp-story animations list.](/documentation/examples/visual-effects/amp_story_animations/?format=stories#zoom-in)
 
     Tip: It's important that the **image** sets the **panning animation**, whereas the image **container** sets the **zoom animation**. Otherwise the dimensions of the image will change, causing undesired results in the animation.
+    
+    Tip: Animations can help make your Web Story more visually exciting and engaging, but use them sparingly. Some users may find long, continuous animations distracting. Other users may have motion sensitivity and be adversely affected by excessive use of motion and parallax effects. This [tracking issue](https://github.com/ampproject/amphtml/issues/24821) follows the effort to respect the `prefers-reduced-motion` directive.
```
[the remark this answers is not in the document]
Comment (caroqliu) [reply examples/source/visual-effects/Ken_Burns.html:67]: Updated wording, PTAL

**Review by mszylkowski — APPROVED** [review]
Perfect, LGTM!

**Pull request merged**